### ISSUE TITLE
v0.12.0 tracking issue

### ISSUE THREAD

**Issue by coreylowman** [coreylowman/dfdx#743]
This is going to be a smaller breaking release focused around f16 and concat/stack. This is to support the development of transformer models.

- [x] #423 
- [x] #731 
- [x] #693 
- [x] #589 
- [ ] #219 
- [ ] #424 

Lower prio, but nice to have:
- [x] #721 
- [ ] #718

**Comment by coreylowman** [comment]
Note: this is currently blocked on a new release of the `half-rs` crate.

**Comment by ViliamVadocz** [comment]
New version of `half` has been released. Is this no longer blocked?

**Comment by coreylowman** [comment]
Yep, and the 0.9.11 version of cudarc uses that, so once I update the conv transpose for dilation/groups I'll push the release. Probably won't add AMP for now
